clang/lib/DPCT/APINamesGraphicsInterop.inc

```diff
@@ -0,0 +1,96 @@
+//===--------------------- APINamesGraphicsInterop.inc --------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// clang-format off
+
+CONDITIONAL_FACTORY_ENTRY(
+    UseExtBindlessImages,
+    ASSIGNABLE_FACTORY(ASSIGN_FACTORY_ENTRY(
+        "cudaGraphicsD3D11RegisterResource", DEREF(0),
+        NEW(MapNames::getDpctNamespace() +
+                "experimental::interop_mem_wrapper",
+            ARG(1), ARG(2)))),
+    UNSUPPORT_FACTORY_ENTRY("cudaGraphicsD3D11RegisterResource",
+                            Diagnostics::TRY_EXPERIMENTAL_FEATURE,
+                            ARG("cudaGraphicsD3D11RegisterResource"),
+                            ARG("--use-experimental-features=bindless_images")))
+
+CONDITIONAL_FACTORY_ENTRY(
+    UseExtBindlessImages,
+    CONDITIONAL_FACTORY_ENTRY(
+        checkIsCallExprOnly(),
+        WARNING_FACTORY_ENTRY("cudaGraphicsResourceSetMapFlags",
+            TOSTRING_FACTORY_ENTRY("cudaGraphicsResourceSetMapFlags", LITERAL("")),
+            Diagnostics::FUNC_CALL_REMOVED,
+            std::string("cudaGraphicsResourceSetMapFlags"), getRemovedAPIWarningMessage("cudaGraphicsResourceSetMapFlags")),
+        WARNING_FACTORY_ENTRY("cudaGraphicsResourceSetMapFlags",
+            TOSTRING_FACTORY_ENTRY("cudaGraphicsResourceSetMapFlags", LITERAL("0")),
+            Diagnostics::FUNC_CALL_REMOVED_0,
+            std::string("cudaGraphicsResourceSetMapFlags"), getRemovedAPIWarningMessage("cudaGraphicsResourceSetMapFlags"))),
+    UNSUPPORT_FACTORY_ENTRY("cudaGraphicsResourceSetMapFlags", Diagnostics::TRY_EXPERIMENTAL_FEATURE,
```

Comment:
It seems like for the removed API rule, we need not check whether the bindless images extension is enabled. Because se not use the bindless image API at all.